File: src/015/p015.ts
const factorial = (n: number): number => {
  let fac = 1;
  for (let i = 1; i <= n; i++) {
    fac *= i;
  }
  return fac;
};

export const solve = (): string => {
  const n = 20;
  return (factorial(2 * n) / (factorial(n) * factorial(n))).toString();
};

File: src/015/p015.go
package problem015

import (
	"math/big"

	"github.com/sunilbpandey/project-euler/src/utils/go/bigintmath"
)

func Solve() string {
	fac20 := bigintmath.Factorial(20)
	fac40 := bigintmath.Factorial(40)

	fac20Squared := new(big.Int).Mul(fac20, fac20)
	answer := new(big.Int).Div(fac40, fac20Squared)
	return answer.String()
}
